### PR TITLE
Support Secured - Claims using TotalEligible Amount 

### PR DESCRIPTION
**Pull request by anandrgitnirman**
- Fixed all existing test cases , Added integration test cases for really big amounts on Rewards 
- Added new methods for the more secure claim signatures
- Added total_eligibile_amount  , this needs to be passed additionally

### CODE REVIEW

File: airdrop/application/services/airdrop_services.py [hunk 1]
```diff
@@ -257,6 +257,45 @@ def airdrop_window_claim_status(self, inputs):
...
+            signature = self.get_signature_for_airdrop_window_id_with_totaleligibilty_amount(
+                claimable_amount, total_eligibility_amount,airdrop_id, airdrop_window_id, user_wallet_address, contract_address, token_address)
+
+            response = AirdropClaim(airdrop_id,
+                                    airdrop_window_id, user_wallet_address, signature, claimable_amount, token_address, contract_address, staking_contract_address,total_eligibility_amount).to_dict()
+
+            status = HTTPStatus.OK
+        except ValidationError as e:
+            response = e.message
+        except BaseException as e:
```
Comment (raamb) [inline airdrop/application/services/airdrop_services.py:292]: Use logger

File: airdrop/infrastructure/models.py [hunk 2]
```diff
@@ -170,7 +171,7 @@ class UserRewardAudit(Base, AuditClass):
     staked = Column("staked", BIGINT, nullable=False)    
     score = Column("score", DECIMAL(18,8), nullable=False)
     normalized_score = Column("normalized_score", DECIMAL(18,8), nullable=False)
-    rewards_awarded = Column("rewards_awarded", BIGINT, nullable=False)
+    rewards_awarded = Column("rewards_awarded", VARCHAR(50), nullable=False)
```
Comment (ksridharbabuus) [inline airdrop/infrastructure/models.py:174]: Can you please make sure that ant existing Queries with the Aggregated functions like sum are not used on this data type.

Comment (anandrgitnirman) [reply airdrop/infrastructure/models.py:174]: ```DESCRIBE user_rewards;```
**O/P**
```
Field            |Type         |Null|Key|Default          |Extra                                        |
-----------------+-------------+----+---+-----------------+---------------------------------------------+
row_id           |bigint       |NO  |PRI|                 |auto_increment                               |
row_created      |timestamp    |NO  |   |CURRENT_TIMESTAMP|DEFAULT_GENERATED                            |
row_updated      |timestamp    |NO  |   |CURRENT_TIMESTAMP|DEFAULT_GENERATED on update CURRENT_TIMESTAMP|
airdrop_id       |bigint       |NO  |MUL|                 |                                             |
airdrop_window_id|bigint       |NO  |MUL|                 |                                             |
address          |varchar(50)  |NO  |MUL|                 |                                             |
condition        |text         |YES |   |                 |                                             |
rewards_awarded  |varchar(50)  |NO  |   |                 |                                             |
normalized_score |decimal(10,8)|NO  |   |                 |                                             |
score            |decimal(10,8)|NO  |   |                 |                                             |
```

2nd Query 

```
select * from user_rewards where address = '0xCc3cD60FF9936B7C9272a649b24f290ADa562469';
```
O/P
```
row_id|row_created        |row_updated        |airdrop_id|airdrop_window_id|address                                   |condition|rewards_awarded        |normalized_score|score     |
------+-------------------+-------------------+----------+-----------------+------------------------------------------+---------+-----------------------+----------------+----------+
    40|2021-12-28 15:03:33|2022-01-14 13:14:22|         1|                7|0xCc3cD60FF9936B7C9272a649b24f290ADa562469|         |10000000000000000000000|      0.00000000|0.00000000|
    41|2021-12-28 15:03:33|2022-01-14 13:14:23|         1|                8|0xCc3cD60FF9936B7C9272a649b24f290ADa562469|         |10000000000000000000000|      0.00000000|0.00000000|
    42|2021-12-28 15:03:33|2022-01-14 13:14:23|         1|                9|0xCc3cD60FF9936B7C9272a649b24f290ADa562469|         |10000000000000000000000|      0.00000000|0.00000000|
    46|2021-12-28 15:03:33|2022-01-14 13:11:33|         2|                6|0xCc3cD60FF9936B7C9272a649b24f290ADa562469|         |10000000000000000000000|      0.00000000|0.00000000|
```

3rd Query 
```
select  sum(ur.rewards_awarded) as 'total_rewards' from user_rewards ur where ur.address = '0xCc3cD60FF9936B7C9272a649b24f290ADa562469';
```
O/P

```
total_rewards          |
-----------------------+
40000000000000000000000|

```

Comment (anandrgitnirman) [reply airdrop/infrastructure/models.py:174]: WE use SQL to sum total_rewards  and total_eligibility , functioal test case , check this piece for aggrefated sums as well

**Review by ksridharbabuus — APPROVED** [review]
Approved.

**Review by raamb — APPROVED** [review]

**Pull request merged**